### PR TITLE
Simplifies handleOrProxyAll interface by making it async instead of sync

### PR DESCRIPTION
**Pull request by markyen**
@jwolski @Raynos

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -871,9 +880,19 @@ RingPop.prototype.handleOrProxyAll =
                     dest: dest
                 }));
             }
-            responses[dest] = res;
-            return responses;
-        }, {});
+        });
+
+        function onResponse(err, data, dest) {
+            if (data) {
```
Comment (Raynos) [inline index.js]: you cant just add a dest & keys field to data. You want `{ body: data, dest, keys }`

Comment (markyen) [inline index.js]: Fair enough

File: index.js [hunk 2]
```diff
@@ -843,37 +843,56 @@ RingPop.prototype.handleOrProxy =
     };
 
 RingPop.prototype.handleOrProxyAll =
-    function handleOrProxyAll(opts) {
+    function handleOrProxyAll(opts, cb) {
```
Comment (jwolski) [inline index.js]: lgtm, but to be comprehensive make sure that `cb()` is called even with 0 keys?

**Comment by Raynos** [comment]
lgtm.

File: index.js (hunk 2 above)
Comment (markyen) [inline index.js]: Good catch!

**Pull request merged**